engine/src/main/java/org/terasology/math/Region3i.java

```diff
@@ -73,253 +223,676 @@ public static Region3i createFromMinAndSize(BaseVector3i min, BaseVector3i size)
+     *             Use the JOML implementation instead: {@link #inflate(int)} or {@link #inflate(int, Region3i)}
+     *             Note: {@link #inflate(int)} will modify <code>this</code>
      */
+    @Deprecated
     public Region3i expand(int amount) {
-        Vector3i expandedMin = min();
+        org.terasology.math.geom.Vector3i expandedMin = min();
         expandedMin.sub(amount, amount, amount);
-        Vector3i expandedMax = max();
+        org.terasology.math.geom.Vector3i expandedMax = max();
         expandedMax.add(amount, amount, amount);
         return createFromMinMax(expandedMin, expandedMax);
     }
 
+    /**
+     *
+     * @param amount
+     * @return A new region
+     * @deprecated This method is scheduled for removal in an upcoming version.
+     *             Use the JOML implementation instead: {@link #inflate(int, Region3i)}
+     */
+    @Deprecated
     public Region3i expand(BaseVector3i amount) {
-        Vector3i expandedMin = min();
+        org.terasology.math.geom.Vector3i expandedMin = min();
         expandedMin.sub(amount);
-        Vector3i expandedMax = max();
+        org.terasology.math.geom.Vector3i expandedMax = max();
         expandedMax.add(amount);
         return createFromMinMax(expandedMin, expandedMax);
     }
 
+    /**
+     * @param adjPos
+     * @return
+     * @deprecated This method is scheduled for removal in an upcoming version.
+     *             Use the JOML implementation instead: {@link #inflate(int, int, int, Region3i)}
+     */
+    @Deprecated
     public Region3i expandToContain(BaseVector3i adjPos) {
-        Vector3i expandedMin = min();
+        org.terasology.math.geom.Vector3i expandedMin = min();
         expandedMin.min(adjPos);
-        Vector3i expandedMax = max();
+        org.terasology.math.geom.Vector3i expandedMax = max();
         expandedMax.max(adjPos);
         return createFromMinMax(expandedMin, expandedMax);
     }
 
     /**
      * @return The position at the center of the region
+     * @deprecated This method is scheduled for removal in an upcoming version.
+     *             Use the JOML implementation instead: {@link #getCenter(Vector3f)}.
      */
-    public Vector3f center() {
-        Vector3f result = min.toVector3f();
-        Vector3f halfSize = size.toVector3f();
+    @Deprecated
+    public org.terasology.math.geom.Vector3f center() {
+        org.terasology.math.geom.Vector3f result = new org.terasology.math.geom.Vector3f(minX, minY, minZ);
+        org.terasology.math.geom.Vector3f halfSize = new org.terasology.math.geom.Vector3f(sizeX(), sizeY(), sizeZ());
         halfSize.scale(0.5f);
         result.add(halfSize);
         return result;
     }
 
+    /**
+     * get center of {@link Region3i} and write to <code>dest</code>
+     *
+     * @param dest will hold the result
+     * @return dest
+     */
+    public Vector3f getCenter(Vector3f dest) {
```

Comment:
You probably intended to use the JOML-type here. You are still importing TeraMath.Vector3f and are therefore using it in this method.

Reply:
looks like I overlooked the import.